src/controllers/Social.ts

```diff
@@ -419,6 +407,44 @@ export default class Social {
     );
   }
 
+  /**
+   * Sets the new social contract ID.
+   * @param {string} contractId - the account of the new social contract ID.
+   */
+  public setContractId(contractId: string): void {
+    this.contractId = contractId;
+  }
+
+  /**
+   * Gets the storage balance for a given account ID.
+   * @param {IStorageBalanceOfOptions} options - the account ID to check and the signer that is used to call the view
+   * method.
+   * @returns {Promise<IStorageBalanceOfResult | null>} a promise that resolves the total & available balance or null
+   * if the account ID does not have a balance.
+   */
+  public async storageBalanceOf({
+    accountId,
+    signer,
+  }: IStorageBalanceOfOptions): Promise<IStorageBalanceOfResult | null> {
+    const result: ISocialDBContractStorageBalance | null =
+      await signer.viewFunction({
```

Comment:
We should use the new rpcQueries utility here for view function and eliminate the need of a signer for storageBalanceOf()